File: DSA/SuryanshMishra_590012069/Day62/day62question3.cpp
#include <vector>
#include <queue>
using namespace std;
long long kthLargestSubarraySum(const vector<int>& arr, int k) {
	int n = (int)arr.size();
	if (k <= 0) return 0;
	vector<long long> pref(n + 1, 0);
	for (int i = 0; i < n; ++i) pref[i + 1] = pref[i] + arr[i];
	priority_queue<long long, vector<long long>, greater<long long>> minHeap;
	for (int i = 0; i < n; ++i) {
		for (int j = i; j < n; ++j) {
			long long sum = pref[j + 1] - pref[i];
			if ((int)minHeap.size() < k) {
				minHeap.push(sum);
			} else if (sum > minHeap.top()) {
				minHeap.pop();
				minHeap.push(sum);
			}
		}
	}
	return minHeap.empty() ? 0 : minHeap.top();
}

File: DSA/VenikaPanwar_590013755/Day_62_Question_3.java
import java.util.*;

public class KthLargestSubarraySum {
    public static int kthLargestSum(int[] arr, int k) {
        int n = arr.length;
        int[] prefix = new int[n + 1];
        
        for (int i = 1; i <= n; i++) prefix[i] = prefix[i - 1] + arr[i - 1];

        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        for (int start = 0; start < n; start++) {
            for (int end = start; end < n; end++) {
                int sum = prefix[end + 1] - prefix[start];
                minHeap.add(sum);
                if (minHeap.size() > k) minHeap.poll();
            }
        }
        return minHeap.peek();
    }

    public static void main(String[] args) {
        System.out.println(kthLargestSum(new int[]{10, -10, 20, -40}, 3)); // 10
        System.out.println(kthLargestSum(new int[]{1, 2, 3}, 2));          // 5
    }
}
